secure-api-gateway-ob-uk-rs-server/src/main/java/com/forgerock/sapi/gateway/ob/uk/rs/server/api/admin/data/FakeDataApiController.java

```diff
@@ -368,6 +373,7 @@ private FRBalance generateBalance(com.forgerock.sapi.gateway.ob.uk.rs.server.per
         Double amount = generateAmount(1000.0d, 10000.0d);
         FRBalance balance = new FRBalance();
         balance.setAccountId(account.getId());
+        balance.setId(account.getId());
```

Comment:
is this needed?

Reply:
It's the unique key of balance and could be an UUID, because we use `findByAccountId` but could be helpful `findById` using the accountId if necessary, at last the account Id it's unique too.

Reply:
I'd rather we not rely on ids matching like this. Using the UUID approach that we had before seems fine to me.

The join is on account.id == balance.accountId , so we just need to query balance.accountId (or do a join).

Put another way, if we don't make this change, then does it work?